Marlin/SanityCheck.h

```diff
@@ -261,6 +261,15 @@
 #endif
 
 /**
+ * DHT Sensor option
+ */
+#if ENABLED(DHT_ENABLE)
+  #if DISABLED(ULTIPANEL)
+    #error "DHT Sensor option requires an LCD controller."
+  #endif
+#endif
```

Comment:
Yes...   Sanity.cpp would be the best place for this check.   But I presume you would be putting a:
```cpp
#define DHT_ENABLE
```
in front of that check, right?